sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsClientTestBase.java

```diff
@@ -0,0 +1,1014 @@
+        ));
+    }
+
+    void analyseBatchSentimentShowStatsRunner(
+        BiConsumer<List<TextDocumentInput>, TextAnalyticsRequestOptions> testRunner) {
+        final List<TextDocumentInput> textDocumentInputs = Arrays.asList(
+            new TextDocumentInput("0", "The hotel was dark and unclean. The restaurant had amazing gnocchi."),
+            new TextDocumentInput("1", "The restaurant had amazing gnocchi. The hotel was dark and unclean.")
+        );
+        testRunner.accept(textDocumentInputs, setTextAnalyticsRequestOptions());
+    }
+
+    static List<String> getSentimentInput() {
+        return Arrays.asList("The hotel was dark and unclean. The restaurant had amazing gnocchi.",
+            "The restaurant had amazing gnocchi. The hotel was dark and unclean.");
+    }
+
+    String getEndpoint() {
+        return interceptorManager.isPlaybackMode()
+            ? "http://localhost:8080"
+            : Configuration.getGlobalConfiguration().get("AZURE_TEXT_ANALYTICS_ENDPOINT");
+    }
+
+    /**
+     * Helper method to verify batch result.
+     *
+     * @param actualResult DocumentResultCollection<> returned by the API.
+     * @param testApi the API to test.
+     */
+    <T> void validateBatchResult(DocumentResultCollection<T> actualResult,
+        DocumentResultCollection<T> expectedResult, TestEndpoint testApi) {
+        // assert batch result
+        assertEquals(expectedResult.getModelVersion(), actualResult.getModelVersion());
+        if (this.showStatistics) {
+            validateBatchStatistics(expectedResult.getStatistics(), actualResult.getStatistics());
+        }
+        validateDocuments(expectedResult, actualResult, testApi);
+    }
+
+    /**
+     * Helper method to verify documents returned in a batch request.
+     *
+     * @param expectedResult the expected result collection..
+     * @param actualResult the actual result collection returned by the API.
+     * @param testApi the API to test.
+     */
+    private <T> void validateDocuments(DocumentResultCollection<T> expectedResult,
+        DocumentResultCollection<T> actualResult, TestEndpoint testApi) {
+        switch (testApi) {
+            case LANGUAGE:
+                final List<DetectLanguageResult> detectLanguageResults = expectedResult.stream()
+                    .filter(element -> element instanceof DetectLanguageResult)
+                    .map(element -> (DetectLanguageResult) element)
+                    .collect(Collectors.toList());
+
+                final List<DetectLanguageResult> actualDetectLanguageResults = actualResult.stream()
+                    .filter(element -> element instanceof DetectLanguageResult)
+                    .map(element -> (DetectLanguageResult) element)
+                    .collect(Collectors.toList());
+                assertEquals(detectLanguageResults.size(), actualDetectLanguageResults.size());
+
+                actualDetectLanguageResults.forEach(actualItem -> {
```

Comment:
In my opinion, it is better to compare the expected set against actual, rather than other way around. What if `expected` has more items than `actual`? Then you're missing something.

Reply:
Actually, [here](https://github.com/Azure/azure-sdk-for-java/pull/6161/files#diff-e04067c9d30740b7c67fa86108b17e8dR502) we check if the size of the expected List and actual List is the same. If different this will never be called. Which is also correct, if different sizes are returned it should error.